### PR TITLE
chore: adds bundle size metrics

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

Adds bundle size fixtures 

## Related Issue(s)

- Fixes https://github.com/microsoft/fluentui/issues/28691

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Avatar-mount_base.html">601</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Avatar-mount.html">595</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Button-mount_base.html">291</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Button-mount.html">298</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Field-mount_base.html">1035</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Field-mount.html">1038</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProvider-mount_base.html">645</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProvider-mount.html">663</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProviderWithTheme-mount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProviderWithTheme-mount.html">73</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">72</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">70</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/InfoButton-mount_base.html">10</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/InfoButton-mount.html">20</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/MakeStyles-mount_base.html">851</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/MakeStyles-mount.html">839</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Persona-mount_base.html">1645</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/Persona-mount.html">1559</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/SpinButton-mount_base.html">1292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28692/perf-test-react-components/SpinButton-mount.html">1338</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                               | Baseline (minified/GZIP) |                           PR |       Change |
| :---------------------------------------------------------------------------------------------- | -----------------------: | ---------------------------: | -----------: |
| <samp>react-tree</samp> <br /> <abbr title='bundle-size/FlatTree.fixture.js'>Tree</abbr>        |         `0 B`<br />`0 B` | `77.951 kB`<br />`22.037 kB` | 🆕 New entry |
| <samp>react-tree</samp> <br /> <abbr title='bundle-size/PersonaFlatTree.fixture.js'>Tree</abbr> |         `0 B`<br />`0 B` | `79.314 kB`<br />`22.345 kB` | 🆕 New entry |
| <samp>react-tree</samp> <br /> <abbr title='bundle-size/PersonaTree.fixture.js'>Tree</abbr>     |         `0 B`<br />`0 B` |   `76.1 kB`<br />`21.194 kB` | 🆕 New entry |
| <samp>react-tree</samp> <br /> <abbr title='bundle-size/Tree.fixture.js'>Tree</abbr>            |         `0 B`<br />`0 B` | `74.735 kB`<br />`20.882 kB` | 🆕 New entry |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.576 kB`<br />`18.225 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `206.083 kB`<br />`57.397 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.409 kB`<br />`12.003 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |     `6.48 kB`<br />`2.203 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/937b241d624114c181ea1521d0d4ead1068ddfa3'>937b241d624114c181ea1521d0d4ead1068ddfa3</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-tree-c65bb12d-453d-4ee3-910d-058cfa260cd3.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
```
Comment (layershifter) [inline change/@fluentui-react-tree-c65bb12d-453d-4ee3-910d-058cfa260cd3.json:2]:
```suggestion
  "type": "none",
```

File: packages/react-components/react-tree/bundle-size/FlatTree.fixture.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  // eslint-disable-next-line camelcase
+  useHeadlessFlatTree_unstable,
+  TreeItem,
+  TreeItemLayout,
+} from '@fluentui/react-tree';
+
+console.log(FlatTree, TreeItem, TreeItemLayout, flattenTree_unstable, useHeadlessFlatTree_unstable);
+
+export default {
+  name: 'Tree',
```
Comment (layershifter) [inline packages/react-components/react-tree/bundle-size/FlatTree.fixture.js:14]:
```suggestion
  name: 'FlatTree',
```

File: packages/react-components/react-tree/bundle-size/PersonaFlatTree.fixture.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  // eslint-disable-next-line camelcase
+  useHeadlessFlatTree_unstable,
+  TreeItem,
+  TreeItemPersonaLayout,
+} from '@fluentui/react-tree';
+
+console.log(FlatTree, TreeItem, TreeItemPersonaLayout, flattenTree_unstable, useHeadlessFlatTree_unstable);
+
+export default {
+  name: 'Tree',
```
Comment (layershifter) [inline packages/react-components/react-tree/bundle-size/PersonaFlatTree.fixture.js:14]:
```suggestion
  name: 'PersonaFlatTree',
```

File: packages/react-components/react-tree/bundle-size/PersonaTree.fixture.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { Tree, TreeItem, TreeItemPersonaLayout } from '@fluentui/react-tree';
+
+console.log(Tree, TreeItem, TreeItemPersonaLayout);
+
+export default {
+  name: 'Tree',
```
Comment (layershifter) [inline packages/react-components/react-tree/bundle-size/PersonaTree.fixture.js:6]:
```suggestion
  name: 'PersonaTree',
```

**Review by layershifter — APPROVED** [review]

**Pull request merged**